src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -135,9 +135,34 @@ public async Task<ChainedHeader> InitializeAsync(ChainedHeader highestHeader)
             {
                 // Repository is behind chain of headers.
                 tip = tip.FindAncestorOrSelf(repoTip.Hash, repoTip.Height);
+
+                if (tip == null)
+                {
+                    // Start at one less of the current repo height as we have already checked
+                    // the repo tip.
+                    for (int height = repoTip.Height - 1; height > 0; height--)
+                    {
+                        var provenBlockHeader = await this.provenBlockHeaderRepository.GetAsync(height).ConfigureAwait(false);
+                        if (provenBlockHeader.GetHash() == highestHeader.HashBlock)
```

Comment:
This check is wrong I believe, we need to check on the chain itself `highestHeader.FindAncestorOrSelf(ph-hash)`